File: leetcode2/0766. Toeplitz Matrix.py
class Solution:
    def isToeplitzMatrix(self, matrix: 'List[List[int]]') -> 'bool':
        row=len(matrix)-1
        column=0
        while row>=0 and column<len(matrix[0]):
            if not self.subisToeplitzMatrix(row,column,matrix):
                return False
            else:
                if row!=0:
                    row-=1
                else:
                    column+=1
        return True
    
    def subisToeplitzMatrix(self,row,column,matrix):
        target=matrix[row][column]
        while row+1<len(matrix) and column+1<len(matrix[0]):
            if matrix[row+1][column+1]!=target:
                return False
            else:
                row+=1
                column+=1
        return True

File: archives/C#/0766. Toeplitz Matrix.cs
public class Solution {
    public bool IsToeplitzMatrix(int[,] matrix) {
        for (int row=0;row<matrix.GetLength(0);row++){
            if (IsSameValue(row,0,matrix)==false){
                return false;
            }
        }
        for (int column=0;column<matrix.GetLength(1);column++){
            if (IsSameValue(0,column,matrix)==false){
                return false;
            }
        }
        return true;
    }
    public bool IsSameValue(int row,int column,int[,] matrix){
        int target=matrix[row,column];
        row+=1;
        column+=1;
        while(row<matrix.GetLength(0) && column<matrix.GetLength(1)){
            if (matrix[row,column]!=target){
                return false;
            }
            row++;
            column++;
        }
        return true;
    }
}
